### PR TITLE
Improve Nix support

### PR DESCRIPTION
**Pull request by Lunaphied**
Being able to use Séance under NixOS would be very useful, this is an initial attempt at defining a NixOS service.

### Path to completion
- [x] Add service definition module for NixOS.
- [x] Update the documentation with Nix usage instructions

### CODE REVIEW

**Comment by Lunaphied** [comment]
I've updated this to use flakes in a basic capacity and cleaned up the derivation for Séance itself, however a solution needs to be found to work around the issues using nix-shell since the previous workaround is predictably too fragile. 

This seems to address a similar/identical issue but for Poetry based projects using a similar technique. https://github.com/nix-community/poetry2nix/pull/433

**Comment by Lunaphied** [comment]
This should implement all the required functionality now

**Comment by Lunaphied** [comment]
Converting this back into a draft because it both needs some cleanups, and also will need to be adapted a bit if/when Séance gets an overhaul. I'll do the cleanups and mark it as ready though, since it might be awhile before that happens

**Comment by Lunaphied** [comment]
@ktemkin I think you have a patch that supersedes this, are you planning on merging that or should I update this and put it for review?

EDIT: Resolved elsewhere, I'm going to pull in some changes discussed

**Comment by Lunaphied** [comment]
This should be ready for review now @Qyriad

File: nix/module.nix [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+        type = types.bool;
+        default = false;
+        description = ''
+          Enables Seance servers for each specified user.
+        '';
+      };
+
+      package = mkOption {
+        type = types.path;
+        default = self.packages.${pkgs.system}.default;
```
Comment (Qyriad) [inline nix/module.nix]: Is there a reason to take all of `self`, which is only used here, instead of just taking the package in the first place?

**Review by Qyriad — APPROVED** [review]

**Pull request merged**